docs/examples/pytorch/single_stage_detector/src/coco_pipeline.py

```diff
@@ -12,96 +12,105 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-import torch
-import ctypes
-import logging
-
-import numpy as np
 
-# DALI imports
+import torch
 from nvidia.dali.pipeline import Pipeline
 import nvidia.dali.ops as ops
 import nvidia.dali.types as types
 
-import time
-
-
 class COCOPipeline(Pipeline):
-    def __init__(
-        self, 
-        batch_size,  
-        file_root, 
-        annotations_file, 
-        default_boxes,
-        seed,
-        device_id=0,
-        num_threads=4):
-
+    def __init__(self, default_boxes, args, seed):
         super(COCOPipeline, self).__init__(
-            batch_size=batch_size, device_id=device_id, num_threads=num_threads, seed = seed)
+            batch_size=args.batch_size, 
+            device_id=args.local_rank,
+            num_threads=args.num_workers, 
+            seed = seed)
+
+        try:
+            shard_id = torch.distributed.get_rank()
+        except RuntimeError:
+            shard_id = 0
 
         self.input = ops.COCOReader(
-            file_root = file_root,
-            annotations_file = annotations_file,
-            ratio=True, ltrb=True,
-            random_shuffle=True)
-        self.decode = ops.nvJPEGDecoder(device="mixed", output_type=types.RGB)
+            file_root = args.train_coco_root, 
```

Comment:
This file has very inconsistent white spaces when passing keyword arguments.

For example `file_root = args.train_coco_root,` and `skip_empty=True,`